src/darwin/Framework/CHIP/MTRDeviceControllerDataStore.h

```diff
@@ -48,6 +48,13 @@ NS_ASSUME_NONNULL_BEGIN
                             storageDelegate:(id<MTRDeviceControllerStorageDelegate>)storageDelegate
                        storageDelegateQueue:(dispatch_queue_t)storageDelegateQueue;
 
+// clusterDataByNode a dictionary: nodeID => cluster data dictionary
+typedef void (^MTRDeviceControllerDataStoreClusterDataHandler)(NSDictionary<NSNumber *, NSDictionary<MTRClusterPath *, MTRDeviceClusterData *> *> * clusterDataByNode);
+/**
+ * Asks the data store to load cluster data for nodes in bulk, if the storageDelegate supports it.
```

Comment:
What's the contract around this?  Will clusterDataHandler not be called at all if the storageDelegate does not support bulk reads?  Something else?

Is clusterDataHandler called sync or async?  Caller certainly seems to asume sync, at least insofar as its lack of locking went.

Reply:
I'll add these details.